feat: log_read min_level filter silent failure on unstructured logs

## Problem

`log_read` accepts a `min_level` parameter (DEBUG, INFO, WARNING, ERROR, CRITICAL) but silently returns 0 results when applied to unstructured logs (Apache, daemon logs) that don't use Jeedom's structured log format.

### Observed behavior

```
log_read(log="http.error", min_level="ERROR")
→ { "total": 0, "lines": [] }   ← silent failure, no warning
```

```
log_read(log="http.error")
→ { "total": 678, "lines": [...] }   ← 678 lines exist
```

The `http.error` log uses Apache format: `[Thu Apr 02 07:29:09 2026] [access_compat:error] ...`
Jeedom structured logs use: `[2026-04-02 09:00:00][ERROR] ...`

The `min_level` filter only parses the Jeedom format, so Apache logs always return 0 when filtered — with no indication that the filter was not applied.

## Proposed fix: B + C

### B — Add `format` hint in `logs_list`

Add a `format` field to each entry in `logs_list` so the agent can determine upfront whether `min_level` will be effective:

```json
[
  { "name": "http.error",  "size": 118943, "format": "raw" },
  { "name": "scenario",    "size": 5677,   "format": "jeedom" },
  { "name": "openzwaved",  "size": 5785,   "format": "jeedom" }
]
```

Possible values: `"jeedom"` (structured, supports `min_level`), `"raw"` (unstructured, `min_level` not supported).

### C — Return warning when min_level is passed on an unstructured log

When `min_level` is provided but the log is detected as unstructured, include a `warning` in the response instead of silently returning empty results:

```json
{
  "log": "http.error",
  "total": 678,
  "warning": "min_level filter not applied — unstructured log format (raw). Use search parameter for text filtering instead.",
  "lines": [...]
}
```

## Value

- An AI agent currently cannot distinguish "no errors in log" from "filter not applicable" — both return `total: 0, lines: []`
- `format` hint in `logs_list` allows the agent to choose the right filtering strategy before calling `log_read`
- Warning in response prevents silent failure from being interpreted as a meaningful result

Identified during MCP validation session (2026-04-02).

--- a/api/mcp.php
+++ b/api/mcp.php
@@ -588,7 +588,7 @@ function mcp_get_tools(): array {
         ],
         [
             'name'        => 'logs_list',
-            'description' => 'List available Jeedom log files with their size and last modification date.',
+            'description' => 'List available Jeedom log files with their size, last modification date, format ("jeedom" = structured, supports min_level; "raw" = unstructured, use search instead), and highest log level seen.',
             'inputSchema' => ['type' => 'object', 'properties' => new stdClass(), 'required' => []],
         ],
         [
@@ -1611,6 +1611,23 @@ function log_max_level(string $path): ?string {
     return $max_name;
 }
 
+function log_detect_format(string $path): string {
+    $handle = fopen($path, 'r');
+    if (!$handle) return 'raw';
+    $checked = 0;
+    while (($line = fgets($handle)) !== false && $checked < 5) {
+        $line = trim($line);
+        if ($line === '') continue;
+        if (preg_match('/^\[\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\]\[/', $line)) {
+            fclose($handle);
+            return 'jeedom';
+        }
+        $checked++;
+    }
+    fclose($handle);
+    return 'raw';
+}
+
 function tool_logs_list(): array {
     acl_check('admin_logs', 'read');
     $result = [];
@@ -1620,6 +1637,7 @@ function tool_logs_list(): array {
         if (file_exists($path)) {
             $item['size']      = filesize($path);
             $item['modified']  = date('Y-m-d H:i:s', filemtime($path));
+            $item['format']    = log_detect_format($path);
             $item['max_level'] = log_max_level($path);
         }
         $result[] = $item;
@@ -1634,6 +1652,12 @@ function tool_log_read(string $log, int $lines = 100, int $offset = 0, ?string $
     $path = log::getPathToLog($log);
     if (!file_exists($path)) throw new Exception("Log '{$log}' not found");
     $min_rank = ($min_level !== null) ? ($LOG_LEVELS[strtoupper($min_level)] ?? 0) : -1;
+    $format   = log_detect_format($path);
+    $warning  = null;
+    if ($min_rank >= 0 && $format === 'raw') {
+        $warning  = "min_level filter not applied — unstructured log format (raw). Use the search parameter for text filtering instead.";
+        $min_rank = -1;
+    }
     $all = file($path, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
     if ($min_rank >= 0) {
         $all = array_values(array_filter($all, function ($line) use ($min_rank) {
@@ -1652,12 +1676,14 @@ function tool_log_read(string $log, int $lines = 100, int $offset = 0, ?string $
     $total = count($all);
     // Paginate from the end: offset skips lines from the end, then take $lines
     $slice = array_values(array_slice($all, -($offset + $lines), $lines));
-    return [
+    $result = [
         'log'    => $log,
         'total'  => $total,
         'offset' => $offset,
-        'lines'  => $slice,
     ];
+    if ($warning !== null) $result['warning'] = $warning;
+    $result['lines'] = $slice;
+    return $result;
 }
 
 function tool_updates_list(): array {

